Create 04-60-404-on-istio-gateway.md (#333)
kyma-project/istio: docs/user/04-60-404-on-istio-gateway.md

> # Kyma endpoint returns not found error (404 status code)

Reviewer: ```suggestion
# Kyma endpoint returns a `not found` error (404 status code)
```

## Symptom

> You are accessing a Kyma endpoint at it reports 404 error.

Reviewer: ```suggestion
When you try to access a Kyma endpoint, it reports a 404 error.
```

## Cause

> This might be cause by Istio gateway host conflicts. For example if you create two gateways with the same host, Istio ingress gateway will not be able to deterministically match incoming request to a Gateway, and it will result with requests getting 404 errors. The behaviour is described in https://istio.io/latest/docs/ops/common-problems/network-issues/#404-errors-occur-when-multiple-gateways-configured-with-same-tls-certificate. Keep in mind that creating `Ingress` resources using Istio as their ingress class will also create a `Gateway` entry underneath. To check the cluster configuration you can use `istioctl x internal-debug configz` command.

Reviewer: ```suggestion
The error might be caused by conflicts in the Istio Gateway host. For example, if you create two Gateways with the same host, Istio Ingress Gateway cannot reliably match an incoming request to a specific Gateway. As a result, requests receive 404 errors. Read the [Istio documentation](https://istio.io/latest/docs/ops/common-problems/network-issues/#404-errors-occur-when-multiple-gateways-configured-with-same-tls-certificate) to learn more about this behavior. 

Note that when you create `Ingress` resources using Istio as their ingress class, a `Gateway` entry is also created underneath. Use the `istioctl x internal-debug configz` command to check the cluster's configuration.
```

## Remedy

> Make sure that a host matches one `Gateway` only.

Reviewer: ```suggestion
Make sure that a host matches only one Gateway.
```
